server/controllers/user.js

```diff
@@ -21,3 +21,35 @@ exports.searchUsers = asyncHandler(async (req, res, next) => {
 
   res.status(200).json({ users: users });
 });
+
+// @route GET /users/:username
+// @desc Get a user by username
+// @access Private
+exports.getUser = asyncHandler(async (req, res, next) => {
+    const username = req.params.username
+    const user = await User.findOne({ username });
+
+    if (!user) res.status(404).send({ error: "User not found" });
+    else res.status(200).send(user);
+});
+
+// @route PATCH /users/:id
+// @desc Update user
+// @access Private
+exports.updateUser = asyncHandler(async (req, res, next) => {
+    const updates = Object.keys(req.body);
+    const allowedUpdates = ['username', 'email', 'password'];
+    const isValidOperation = updates.every((update) => allowedUpdates.includes(update));
+    if (!isValidOperation) return res.status(404).send({ error: 'Invalid update' });
```

Comment:
Might be best just to set our schemas to strict which will make this kind of logic not necessary as Mongoose will just ignore fields not present in the schema.